datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -944,6 +959,100 @@ private void handleLeaderDoAssignment() {
     }
   }
 
+  /**
+   * assign the partition to tasks for a particular datastreamGroup
+   *
+   * @param maybeDatastreamGroupName the datastreamGroup that needs to perform the partition assignment
+   */
+  private void performPartitionAssignment(Optional<String> maybeDatastreamGroupName) {
+    if (!maybeDatastreamGroupName.isPresent()) {
+      _log.error("Datastream group is not found when performing partition assignment");
+      return;
+    }
+
+    boolean succeeded = false;
+    Map<String, Set<DatastreamTask>> previousAssignmentByInstance = new HashMap<>();
+    Map<String, List<DatastreamTask>> newAssignmentsByInstance = new HashMap<>();
+
+    String datastreamGroupName = maybeDatastreamGroupName.get();
+
+    StickyPartitionAssignmentStrategy partitionAssignmentStrategy = new StickyPartitionAssignmentStrategy();
+
+    try {
+      previousAssignmentByInstance = _adapter.getAllAssignedDatastreamTasks();
+      Map<String, Set<DatastreamTask>> assignmentByInstance = new HashMap<>(previousAssignmentByInstance);
+
+      // retrieve the datastreamGroups for validatio
+      DatastreamGroup toProcessDatastream =
+          fetchDatastreamGroups().stream().filter(dg -> datastreamGroupName.equals(dg.getName())).findFirst().orElse(null);
+
+      if (toProcessDatastream != null) {
```

Comment:
It might happen that the Datastream was paused before the PARTITION_ASSIGNMENT is executed, in that case, the action can be safely ignored. I will add extra log on this. Thanks!